server/util.go

```diff
@@ -254,7 +254,7 @@ func dumpBinaryRow(buffer []byte, columns []*ColumnInfo, row types.Row) ([]byte,
 		case mysql.TypeDouble:
 			buffer = dumpUint64(buffer, math.Float64bits(row.GetFloat64(i)))
 		case mysql.TypeNewDecimal:
-			buffer = dumpLengthEncodedString(buffer, hack.Slice(row.GetMyDecimal(i).String()))
+			buffer = dumpLengthEncodedString(buffer, hack.Slice(string(row.GetMyDecimal(i).ToString())))
```

Comment:
buffer = dumpLengthEncodedString(buffer, row.GetMyDecimal(i).ToString())

Reply:
good idea~finally we revert it and use continue call round up.- -